openfe/protocols/openmm_md/plain_md_methods.py

```diff
@@ -0,0 +1,464 @@
+            protein_comp=protein_comp,
+            solvent_comp=solvent_comp,
+            small_mols=small_mols,
+            omm_forcefield=system_generator.forcefield,
+            solvent_settings=solvation_settings,
+        )
+
+        # d. get topology & positions
+        # Note: roundtrip positions to remove vec3 issues
+        stateA_topology = stateA_modeller.getTopology()
+        stateA_positions = to_openmm(
+            from_openmm(stateA_modeller.getPositions())
+        )
+
+        # e. create the stateA System
+        stateA_system = system_generator.create_system(
+            stateA_topology,
+            molecules=[s.to_openff() for s in small_mols],
+        )
+
+        # 10. Get platform
+        platform = compute.get_openmm_platform(
+            protocol_settings.engine_settings.compute_platform
+        )
+
+        # 11. Set the integrator
+        integrator = openmm.LangevinMiddleIntegrator(
+            to_openmm(thermo_settings.temperature),
+            to_openmm(integrator_settings.collision_rate),
+            to_openmm(integrator_settings.timestep),
+        )
+
+        simulation = openmm.app.Simulation(
+            stateA_modeller.topology,
+            stateA_system,
+            integrator,
+            platform=platform
+        )
+
+        try:
+
+            if not dry:  # pragma: no-cover
+
+                simulation.context.setPositions(stateA_positions)
+
+                # minimize
+                if verbose:
+                    logger.info("minimizing systems")
+
+                simulation.minimizeEnergy(
+                    maxIterations=sim_settings.minimization_steps
+                )
+                min_pdb_file = "minimized.pdb"
+                positions = simulation.context.getState(
+                    getPositions=True, enforcePeriodicBox=False
+                ).getPositions()
+                with open(shared_basepath / min_pdb_file, "w") as f:
+                    openmm.app.PDBFile.writeFile(
+                        simulation.topology, positions, file=f, keepIds=True
+                    )
+                # equilibrate
+                if verbose:
+                    logger.info("equilibrating systems")
+                simulation.context.setVelocitiesToTemperature(
+                    to_openmm(thermo_settings.temperature))
+                simulation.step(equil_steps)
+
+                # production
+                if verbose:
+                    logger.info("running production phase")
+
+                # Get the sub selection of the system to print coords for
+                selection_indices = mdtraj.Topology.from_openmm(
+                    stateA_topology).select(sim_settings.output_indices)
+
+                # Setup the reporters
+                simulation.reporters.append(XTCReporter(
+                    file=str(shared_basepath / sim_settings.output_filename),
+                    reportInterval=sim_settings.checkpoint_interval.m,
```

Comment:
Is `checkpoint_interval` the right setting name for writing out a frame?

It might be that the simulation_settings needs ammending a bit.

Reply:
Also do we want timing info for all the other steps?